WireUI/Sources/WireUIFoundation/Main/MainCoordinator.swift

```diff
@@ -0,0 +1,364 @@
+        switch mainSplitViewState {
+        case .collapsed: tabBarController.conversations?.conversationList
+        case .expanded: splitViewController.conversationList
+        }
+    }
+
+    private var archive: TabBarController.Archive! {
+        switch mainSplitViewState {
+        case .collapsed: tabBarController.archive
+        case .expanded: splitViewController.archive
+        }
+    }
+
+    private var settings: TabBarController.Settings! {
+        switch mainSplitViewState {
+        case .collapsed: tabBarController.settings
+        case .expanded: splitViewController.settings
+        }
+    }
+
+    // MARK: - Life Cycle
+
+    public init(
+        mainSplitViewController: SplitViewController,
+        mainTabBarController: TabBarController,
+        newConversationBuilder: NewConversationBuilder,
+        selfProfileBuilder: SelfProfileBuilder
+    ) {
+        guard
+            mainSplitViewController.conversationList != nil,
+            mainTabBarController.archive != nil,
+            mainTabBarController.settings != nil
+        else { fatalError("invalid state of container view controllers") }
+
+        splitViewController = mainSplitViewController
+        tabBarController = mainTabBarController
+        self.newConversationBuilder = newConversationBuilder
+        self.selfProfileBuilder = selfProfileBuilder
+    }
+
+    // MARK: - Public Methods
+
+    public func showConversationList(conversationFilter: (some MainConversationFilterRepresentable)?) {
+        defer {
+            // switch to the conversation list tab
+            tabBarController.selectedContent = .conversations
+
+            // apply the filter to the conversation list
+            let mainConversationFilter = conversationFilter?.map()
+            conversationList.conversationFilter = mainConversationFilter.map { .init($0) }
+
+            // set the right menu item in the sidebar
+            switch mainConversationFilter {
+            case .none:
+                sidebar.selectedMenuItem = .init(.all)
+            case .favorites:
+                sidebar.selectedMenuItem = .init(.favorites)
+            case .groups:
+                sidebar.selectedMenuItem = .init(.groups)
+            case .oneOnOne:
+                sidebar.selectedMenuItem = .init(.oneOnOne)
+            }
+        }
+
+        // In collapsed state switching the tab was all we needed to do.
+        guard mainSplitViewState == .expanded else { return }
+
+        dismissArchiveIfNeeded()
+        dismissNewConversationIfNeeded()
+        dismissSettingsIfNeeded()
+        dismissSelfProfileIfNeeded()
+
+        // Move the conversation list from the tab bar controller to the split view controller if needed.
+        if let conversationList = tabBarController.conversations?.conversationList {
+            tabBarController.conversations = nil
+            splitViewController.conversationList = conversationList
+        }
```

Comment:
Just to confirm, `tabBarController.conversations?.conversationList` is a valid state at this point right` - i.e we don't need to log an error etc. Same question for the similar methods